sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamQueryPlanner.java

```diff
@@ -142,6 +165,11 @@ public BeamRelNode convertToBeamRel(String sqlStatement)
   }
 
   private Planner getPlanner() {
-    return Frameworks.getPlanner(config);
```

Comment:
I think we still should call calcite's factory explicitly, no on guarantees there's going to be a constructor with this signature

Reply:
It is possible that in the future Calcite will change the way to initialize its plannerImpl. But I doubt when that happens, even the current `Frameworks.getPlanner(config)` will be changed, so Beam will likely to change the API call at that moment.

Right now since we are using calcite-core 1.17.0, and the signature of constructor is guaranteed to be stable. When we upgrade calcite dependency, we can change in Beam if Calcite changes its plannerImpl's constructor signature.

Reply:
When Calcite changes the API while we're calling it directly, Beam SQL will fail to compile. With reflection approach it will fail in the unit tests the earliest, worst case - at runtime.

Reply:
I am not seeing a difference when we upgrade Calcite dependency. All unit tests should pass anyway.

Reply:
Nobody forces you to run unit tests. And nobody forces you to even have tests for this path. But compiler forces you to only call the existing method.

Reply:
I just updated to "if else" to initialize planner differently by the class name.